Core.Auditing/IfxAuditLogger.cs

```diff
@@ -1,101 +1,124 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 
-using Microsoft.Cloud.InstrumentationFramework;
 using Microsoft.CloudMine.Core.Telemetry;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Audit.Geneva;
 using System;
 using System.Collections.Generic;
 using System.Net;
 using System.Net.Sockets;
 
 namespace Microsoft.CloudMine.Core.Auditing
 {
+
+    public class TargetResource
+    {
+        public string type;
+        public string name;
+        public string cluster;
+        public string region;
+
+        public TargetResource(string type, string name, string cluster = null, string region = null)
+        {
+            this.type = type;
+            this.name = name;
+            this.cluster = cluster;
+            this.region = region;
+        }
+    }
+
+    public class CallerIdentity
+    {
+        public CallerIdentityType type;
+        public string name;
+        public string description;
+
+        public CallerIdentity(CallerIdentityType type, string name, string description = null)
+        {
+            this.type = type;
+            this.name = name;
+            this.description = description;
+        }
+    }
+
     public class IfxAuditLogger : IAuditLogger
     {
-        private bool alreadyLoggedPerSession = false;
+        private ILogger logger;
+        private string webAppName;
+        private string ipAddress;
+
         private const string TokenGenerationOperation = "TokenGeneration";
         private const string FetchCertificateOperation = "FetchCertificate";
         private const string DefaultWebAppName = "CloudMinePlatform";
 
-        /// <summary>
-        /// Initializes audit logging.
-        /// </summary>
-        public void Initialize(string tenantIdentity, string roleIdentity)
+        private static readonly AuditLoggerFactory AuditLoggerFactory = AuditLoggerFactory.Create(AuditOptions.DefaultForEtw);
```

Comment:
Is this something that should potentially come through dependency injection?

Reply:
I'd say, if you e.g. expect to use different implementations (or different configuration of the same implementation) in production, local test runs and/or unit tests, choosing DI might be useful.
Or to decouple different pieces of code. Say, if you need this in multiple places, need to ensure there is only one instance, but don't want to couple the different modules to the same static variable.

Reply:
Okay following those guidelines I think it is okay to leave this as a static variable. This object will be identical everywhere (prod / dev / local) and the same instance can be used from anywhere in code.